From f0ba0f9bfa4857a15e0ea06a693ff361124f0a93 Mon Sep 17 00:00:00 2001
From: Vladimir Timofeev <vovkasm@gmail.com>
Date: Tue, 10 May 2011 09:17:41 -0400
Subject: Fix ODBM_File failures on Ubuntu natty [multiarch]

Bug: http://rt.perl.org/rt3/Public/Bug/Display.html?id=90106
Bug-Debian: http://bugs.debian.org/625634
Origin: upstream, http://perl5.git.perl.org/perl.git/commit/3c998e07917e4a06d77c572595b246947120b316

Ubuntu has juggled lib paths so we get to play games to catch up with
them.

Patch-Name: fixes/odbm-multiarch.diff
---
 ext/ODBM_File/hints/linux.pl |    6 +++---
 1 files changed, 3 insertions(+), 3 deletions(-)

diff --git a/ext/ODBM_File/hints/linux.pl b/ext/ODBM_File/hints/linux.pl
index f8ca492..204bba0 100644
--- a/ext/ODBM_File/hints/linux.pl
+++ b/ext/ODBM_File/hints/linux.pl
@@ -1,8 +1,8 @@
 # uses GDBM dbm compatibility feature - at least on SuSE 8.0
 $self->{LIBS} = ['-lgdbm'];
 
-# Debian/Ubuntu have /usr/lib/libgdbm_compat.so.3* but not this file,
+# Debian/Ubuntu have libgdbm_compat.so but not this file,
 # so linking may fail
-if (-e '/usr/lib/libgdbm_compat.so' or -e '/usr/lib64/libgdbm_compat.so') {
-    $self->{LIBS}->[0] .= ' -lgdbm_compat';
+foreach (split / /, $Config{libpth}) {
+    $self->{LIBS}->[0] .= ' -lgdbm_compat' if -e $_.'/libgdbm_compat.so';
 }
